topics/docker.md

```diff
@@ -139,34 +156,50 @@ setx DOCKER_HUB_PASSWORD yourHubPassword
 </tab>
 </tabs>
 
-
 ## Manual image configuration {id="manual"}
 
 If required, you can provide your own `Dockerfile` to assemble an image with a Ktor application.
 
 ### Package the application {id="packagea-pp"}
-As a first step, you need to package your application along with its dependencies. 
-For example, this might be a [fat JAR](server-fatjar.md) or an [executable JVM application](server-packaging.md).
 
+As a first step, you need to package your application along with its dependencies.
+For example, this might be a [fat JAR](server-fatjar.md) or an [executable JVM application](server-packaging.md).
 
 ### Prepare Docker image {id="prepare-docker"}
 
-To dockerize the application, we'll use [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/):
-- First, we'll use the `gradle`/`maven` image to generate a fat JAR with the application.
-- Then, the generated distribution will be run in the environment created based on the `openjdk` image.
+To dockerize the application, we'll
+use [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/):
+
+1. First, we'll set up caching for Gradle/Maven dependencies. This step is optional, but recommended as it improves the
+   overall build speed.
+2. Then, we'll use the `gradle`/`maven` image to generate a fat JAR with the application.
+3. Finally, the generated distribution will be run in the environment created based on the `openjdk` image.
 
 In the root folder of the project, create a file named `Dockerfile` with the following contents:
 
 <tabs group="languages">
 <tab title="Gradle" group-key="kotlin">
 
 ```Docker
-FROM gradle:7-jdk11 AS build
+# Stage 1: Cache Gradle dependencies
```

Comment:
Shouldn't this code be referenced from the `codeSnippets/snippets/tutorial-server-docker-compose/Dockerfile` file?